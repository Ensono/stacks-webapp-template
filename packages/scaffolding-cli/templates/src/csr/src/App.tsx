import {ThemeProvider as MuiThemeProvider} from "@material-ui/core/styles"
import React from "react"
import {Helmet} from "react-helmet"
import {Provider} from "react-redux"
import "./App.css"
import Header from "./components/Header"
import Search from "./components/Search"
import theme from "./constants/theme"
import configureStore from "./state"
import ApiPane from "./containers/home"

// eslint-disable-next-line
const initialState = (window as any).initialReduxState
const store = configureStore(initialState)

const App: React.FC = () => {
    return (
        <>
            <Helmet>
                <meta charSet="utf-8" />
                <title>Amido Stacks CSR</title>
            </Helmet>
            <Provider store={store}>
                <MuiThemeProvider theme={theme}>
                    <Header />
                    <br />
                    <br />
                    <Search />
                    <ApiPane />
                </MuiThemeProvider>
            </Provider>
        </>
    )
}

export default App
