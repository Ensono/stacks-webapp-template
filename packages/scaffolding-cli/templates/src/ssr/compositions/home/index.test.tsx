import "@testing-library/jest-dom/extend-expect"
import {render} from "@testing-library/react"
import React from "react"
import {Provider} from "react-redux"
import configureStore from "redux-mock-store"
import sagaMiddleware from "redux-saga"
import App from "../../pages"

const mockStore = configureStore([sagaMiddleware])
const initialState = {
    getMenus: {
        loading: false,
        error: null,
        menuItems: [],
    },
}
const indexPageText = `Yumido`

test("With React Testing Library page renders tag type <div> with text", () => {
    // Initialize mockstore with empty state
    const store = mockStore(initialState)
    const {getByText} = render(
        <Provider store={store}>
            <App />
        </Provider>,
    )

    expect(getByText(indexPageText)).not.toBeNull()
})

test("With React Testing Library Snapshot renders page", () => {
    const store = mockStore(initialState)
    const {asFragment} = render(
        <Provider store={store}>
            <App />
        </Provider>,
    )

    expect(asFragment()).toMatchSnapshot()
})

test("With React Testing Library page rerenders with hydrate", () => {
    const store = mockStore(initialState)
    const {getByText, rerender} = render(
        <Provider store={store}>
            <App />
        </Provider>,
    )
    expect(getByText(indexPageText)).not.toBeNull()

    // Rerender: Calls render again passing in the original arguments and sets hydrate to true.
    rerender(
        <Provider store={store}>
            <App />
        </Provider>,
    )
    expect(getByText(indexPageText)).not.toBeNull()
})
