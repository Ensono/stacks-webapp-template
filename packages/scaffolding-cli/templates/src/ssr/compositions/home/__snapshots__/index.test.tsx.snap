// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With React Testing Library Snapshot renders page 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root css-1knxyyj MuiGrid-container"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorSecondary mui-fixed MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2"
          style="margin: 0px auto;"
        >
          <a
            class="css-15rvyqs"
          >
            Yumido
          </a>
        </h2>
        <button
          aria-label="create menu button"
          class="MuiButtonBase-root MuiFab-root MuiFab-sizeSmall MuiFab-primary"
          tabindex="0"
          title="Create menu"
          type="button"
        >
          <span
            class="MuiFab-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              data-testid="create_button"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </header>
    <main>
      <br />
      <div
        class="MuiContainer-root MuiContainer-maxWidthSm"
      >
        <div
          class="MuiBox-root MuiBox-root-510"
        >
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
            data-testid="search_btn"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink="false"
              for="search-bar"
              id="search-bar-label"
            >
              Search Menus
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputTypeSearch"
                id="search-bar"
                name="SearchTerm"
                type="search"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-569 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-571"
                >
                  <span>
                    Search Menus
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root css-t7xmz MuiGrid-item"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1"
        >
          Latest menus:
        </h1>
        <h2
          class="MuiTypography-root MuiTypography-h2"
        >
          No results
        </h2>
      </div>
    </main>
  </div>
</DocumentFragment>
`;
